src/v/model/record_batch_reader.h

```diff
@@ -292,6 +298,19 @@ class record_batch_reader final {
 
     std::unique_ptr<impl> release() && { return std::move(_impl); }
 
+    // record batch readers may expose these private flags for testing
+    // purposes
+    struct private_flags {
```

Comment:
What is the advantage of making these bit flags?

Reply:
Compared to?

Compared to a flag byte/int and ORing enums/constants it seems a lot better, less error prone, and less code.

Compared to just bool the only advantage is that its smaller, which is mostly useful (in this case) to prevent suggestions of the flag byte above.

Reply:
I did look around for some better "flags" idiom in C++ but came up empty.